src/plugins/browser-console.ts

```diff
@@ -54,12 +54,31 @@ export class BrowserConsole {
     }
   };
 
+  private pageErrorEventListener = (error: Error) => {
+    if (!this._currentStep) {
+      return;
+    }
+    const { name, index } = this._currentStep;
+    this.messages.push({
+      timestamp: getTimestamp(),
+      text: error.message,
+      type: 'error',
+      step: { name, index },
+      stacktrace: error.stack.toString(),
```

Comment:
pass it as error, we have a global formatError function that handles them properly in the console event - https://github.com/elastic/synthetics/blob/e67a91dacc4c981aeb6b0f9d715f07b059da8f80/src/reporters/json.ts#L486-L495